appvm-scripts/usrbin/qubes-run-x11vnc-pre

```diff
@@ -0,0 +1,117 @@
+#!/bin/bash
+#
+# The Qubes OS Project, http://www.qubes-os.org
+#
+# Copyright (C) 2021  Frédéric Pierret  <frederic.pierret@qubes-os.org>
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+#
+
+# Source Qubes library.
+# shellcheck disable=SC1091
+. /usr/lib/qubes/init/functions
+
+array_min_max() {
+    read -r -a array <<< "$@"
+    max=${array[0]}
+    min=${array[0]}
+
+    for i in ${array[*]}; do
+      (( i > max )) && max=$i
+      (( i < min )) && min=$i
+    done
+    echo "$min"-"$max"
+}
+
+if qsvc guivm-vnc; then
+    # Common part from qubes-run-xorg to setup dummyqbs xorg.org
+    GUI_DOMID="$(qubesdb-read /qubes-gui-domain-xid 2>/dev/null)"
```

Comment:
What if guivm isn't set (none)? I'm not sure if it's allowed to have empty value, but if not, please default to 0. This domain ID will really get access to the screen content of the GUI session, so dom0 seems like a safest fallback (if needs to be something).